### PR TITLE
Resolution extent overlay fixes

### PR DESCRIPTION
**Pull request by dbrandenstein**

### CODE REVIEW

**Comment by dbrandenstein** [comment]
[Info-Box über Auflösung und Anfragerechteck](https://trello.com/c/aIlip82E/1643-info-box-%C3%BCber-aufl%C3%B6sung-und-anfragerechteck)

File: projects/core/src/lib/spatial-references/spatial-reference.service.ts [hunk 1]
```diff
@@ -154,14 +155,15 @@ export class SpatialReferenceService {
             map((dict: SpatialReferenceSpecificationDict) => {
                 const spec = SpatialReferenceSpecification.fromDict(dict);
 
-                proj4.defs(spec.projString);
+                proj4.defs(spec.spatialReference.srsString, spec.projString);
```
Comment (michaelmattig) [inline projects/core/src/lib/spatial-references/spatial-reference.service.ts:158]: the units are shown correctly now, but I noticed that before your change, the background layer for UTM32N e.g. would be just black, and after your change it is now a distorted background layer that does not make much sense to me.

Do you have an idea why this is? Before and after your change the console says "Using fallback background layer for EPSG:32632".

File: projects/core/src/lib/spatial-references/spatial-reference.service.ts [hunk 2]
```diff
@@ -154,14 +155,15 @@ export class SpatialReferenceService {
...
+                const def = proj4.defs(spec.spatialReference.srsString);
                 olProj4Register(proj4);
 
                 olAddProjection(
                     new OlProjection({
                         code: spec.spatialReference.srsString,
                         extent: spec.extent,
-                        units: undefined, // TODO: get units from proj or backend
+                        units: def?.units as Units.Units,
                     }),
```
Comment (michaelmattig) [inline projects/core/src/lib/spatial-references/spatial-reference.service.ts:167]:
```suggestion
                        units: def?.units as Units,
```

File: projects/core/src/lib/spatial-references/spatial-reference.service.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ import {UserService} from '../users/user.service';
 import {get as olGetProjection, addProjection as olAddProjection} from 'ol/proj';
 import {register as olProj4Register} from 'ol/proj/proj4';
 import OlProjection from 'ol/proj/Projection';
+import Units from 'ol/proj/Units';
 import proj4 from 'proj4';
```
Comment (michaelmattig) [inline projects/core/src/lib/spatial-references/spatial-reference.service.ts:11]:
```suggestion
import {Units} from 'ol/proj/Units';
```

File: projects/core/src/lib/spatial-references/spatial-reference.service.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ import {UserService} from '../users/user.service';
 import {get as olGetProjection, addProjection as olAddProjection} from 'ol/proj';
 import {register as olProj4Register} from 'ol/proj/proj4';
 import OlProjection from 'ol/proj/Projection';
+import Units from 'ol/proj/Units';
```
Comment (michaelmattig) [inline projects/core/src/lib/spatial-references/spatial-reference.service.ts:10]:
```suggestion
import {Units} from 'ol/proj/Units';
```

File: projects/core/src/lib/spatial-references/spatial-reference.service.ts [hunk 5]
```diff
@@ -154,14 +155,15 @@ export class SpatialReferenceService {
...
+                proj4.defs(spec.spatialReference.srsString, spec.projString);
+                const def = proj4.defs(spec.spatialReference.srsString);
                 olProj4Register(proj4);
 
                 olAddProjection(
                     new OlProjection({
                         code: spec.spatialReference.srsString,
                         extent: spec.extent,
-                        units: undefined, // TODO: get units from proj or backend
+                        units: def?.units as Units.Units,
```
Comment (michaelmattig) [inline projects/core/src/lib/spatial-references/spatial-reference.service.ts:166]:
```suggestion
                        units: def?.units as Units,
```

Comment (dbrandenstein) [reply projects/core/src/lib/spatial-references/spatial-reference.service.ts]: https://github.com/geo-engine/geoengine-ui/pull/733/commits/6864884a95b9d04c305a1db42ae4eae621558597

File: projects/core/src/lib/spatial-references/spatial-reference.service.ts (hunk 4 above)
Comment (dbrandenstein) [reply projects/core/src/lib/spatial-references/spatial-reference.service.ts]: https://github.com/geo-engine/geoengine-ui/pull/733/commits/6864884a95b9d04c305a1db42ae4eae621558597

**Review by michaelmattig — APPROVED** [review]

**Pull request merged**